src/vs/workbench/contrib/terminal/browser/terminalFindWidget.ts

```diff
@@ -78,4 +78,14 @@ export class TerminalFindWidget extends SimpleFindWidget {
 	protected onFindInputFocusTrackerBlur() {
 		this._findInputFocused.reset();
 	}
+
+	public findFirst() {
+		const instance = this._terminalService.getActiveInstance();
+		if (instance) {
+			if (instance.hasSelection()) {
+				instance.clearSelection();
```

Comment:
Huh, I thought the editor's find was more clever about this and wouldn't re-search from the beginning but apparently that's exactly how it works :+1: